File: 0001-twoSum/cpp/Solution.cpp
/**
 * @file Solution.cpp
 * @author your name (you@domain.com)
 * @brief 
 * @version 0.1
 * @date 2022-03-01
 * @details 1. 两数之和
 * @copyright Copyright (c) 2022
 * 
 */

#include <vector>
#include <unordered_map>
#include <iostream>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hash;
        for(int i = 0; i < nums.size(); i++){
        	if(hash.count(target - nums[i])) return {hash[target - nums[i]], i};
        	hash[nums[i]] = i;
		}
		return {-1, -1};
    }
};

int main(int argc, char const *argv[])
{
    /* code */
    vector<int> nums = {2, 7, 11, 15};
    int target = 9;
    for(auto i : Solution().twoSum(nums, target))
    	std::cout << i << std::endl;
    
    return 0;
}

File: 0001-twoSum/python/Solution.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for key, value in enumerate(nums):
            newvalue = target-value
            if newvalue in nums and nums.index(newvalue)!= key :
                return [key, nums.index(newvalue)]

if __name__ == "__main__":
    nums = [2, 7, 11, 15]
    target = 9
    solution = Solution()
    print(solution.twoSum(nums, target))